From 6026f4bb46beca39975b1edda03c1075a6fad9c6 Mon Sep 17 00:00:00 2001
From: Patrik Jakobsson <patrik.r.jakobsson@gmail.com>
Date: Wed, 22 Mar 2017 00:12:35 +0100
Subject: [PATCH 88/88] dma-buf/sync_file: Add back the u64_to_user_ptr() macro
Patch-mainline: Never, fixup for v4.9 DMA-BUF backport
References: fate#321223

Instead of tainting kernel.h with this macro we just add it where it's
needed.

Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/dma-buf/sync_file.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/dma-buf/sync_file.c b/drivers/dma-buf/sync_file.c
index ec61dad76548..1742dcbfa112 100644
--- a/drivers/dma-buf/sync_file.c
+++ b/drivers/dma-buf/sync_file.c
@@ -26,6 +26,13 @@
 #include <linux/sync_file.h>
 #include <uapi/linux/sync_file.h>
 
+#define u64_to_user_ptr(x) (		\
+{					\
+       typecheck(u64, x);		\
+       (void __user *)(uintptr_t)x;	\
+}					\
+)
+
 static const struct file_operations sync_file_fops;
 
 static struct sync_file *sync_file_alloc(void)
-- 
2.12.0
